### PR TITLE
Bugfix/submission tek rolling period

### PR DESCRIPTION
**Pull request by ioangut**
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

This PR is fixing the bug in the distribution service regarding the computation of the expiration policy. The expiration period was calculated based on the rollingPeriod which can be now variable. This PR is fixing the [jira item](https://jira.itc.sap.com/browse/EXPOSUREBACK-168).

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundler.java [hunk 1]
```diff
@@ -102,7 +104,7 @@ protected void createDiagnosisKeyDistributionMap(Collection<DiagnosisKey> diagno
   private LocalDateTime getExpiryDateTime(DiagnosisKey diagnosisKey) {
     return LocalDateTime
         .ofEpochSecond(diagnosisKey.getRollingStartIntervalNumber() * TEN_MINUTES_INTERVAL_SECONDS, 0, UTC)
-        .plusMinutes(diagnosisKey.getRollingPeriod() * 10L);
+        .plusMinutes(ONE_DAY_INTERVAL_ROLLING_PERIOD * 10L);
```
[the remark this answers is not in the document]
Comment (sorin-stefan-iovita-sap) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundler.java:107]: 👍

**Review by michael-burwig — DISMISSED** [review]

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundler.java [hunk 2]
```diff
@@ -102,7 +102,7 @@ protected void createDiagnosisKeyDistributionMap(Collection<DiagnosisKey> diagno
   private LocalDateTime getExpiryDateTime(DiagnosisKey diagnosisKey) {
     return LocalDateTime
         .ofEpochSecond(diagnosisKey.getRollingStartIntervalNumber() * TEN_MINUTES_INTERVAL_SECONDS, 0, UTC)
-        .plusMinutes(diagnosisKey.getRollingPeriod() * 10L);
+        .plusDays(1L);
```
Comment (EugenM-SAP) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundler.java:105]: Worth adding a statement about why we add 1 day as a comment or javadoc

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundlerExpiryPolicyTest.java [hunk 3]
```diff
@@ -85,4 +85,47 @@ void testLastPeriodOfHourAndSubmissionGreaterDistributionDateTime() {
     bundler.setDiagnosisKeys(diagnosisKeys, LocalDateTime.of(1970, 1, 5, 0, 0));
     assertThat(bundler.getDiagnosisKeysForHour(LocalDateTime.of(1970, 1, 2, 4, 0, 0))).hasSize(10);
   }
+
+  @ParameterizedTest
+  @ValueSource(longs = {0L, 24L, 24L + 2L})
+  void testLastPeriodOfHourAndSubmissionLessThanDistributionDateTimeWithFlexibleRollingPeriod(
```
Comment (EugenM-SAP) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundlerExpiryPolicyTest.java:91]: a shorter method name would be better

Comment (michael-burwig) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/ProdDiagnosisKeyBundlerExpiryPolicyTest.java:91]: I agree but the newly introduced code is conforming to the convention that was present before. Could be a candidate for a future refactoring.

**Review by michael-burwig — APPROVED** [review]

**Pull request merged**